File: v3.1/charmm/template/ALF/G_imp/Entropy2.py
#! /usr/bin/env python

import sys, os
import numpy as np

Nmc=5000000
Nl=20
c=5.5
dl=1.0/Nl

Ndim=int(sys.argv[1])

histogram_a=np.zeros((Nl,Nl))

for i in range(0,Nmc):
  ti=np.random.random_sample((Ndim,))
  unli=np.exp(c*np.sin(np.pi*ti-np.pi/2))
  li=unli/np.sum(unli)

  indi=np.floor(li*Nl).astype('int')
  for j in range(0,Ndim):
    for k in range(j+1,Ndim):
      histogram_a[indi[j],indi[k]]+=1
histogram=histogram_a+histogram_a.T

S_k=np.log(histogram)
G_k=-S_k+np.log(np.mean(histogram))

# plot(dl/2:dl:1,histogram1)
# plot((dl/2):dl:1,G12_k')

np.savetxt('G2_'+str(Ndim)+'.dat',G_k)

File: v3.1/charmm/template/ALF/G_imp/old/Entropy2.m
Nmc=5e6;
Nl=20;
c=5.5;
cutlsum=0.8;
dl=1/Nl;

for Ndim=2:10

    Ndim

    histogram2=zeros([Nl,Nl]);
    offdiag=ones([Ndim,Ndim])-eye(Ndim);

    for i=1:Nmc
        ti=rand([1,Ndim]);
        unli=exp(c*sin(pi*ti-pi/2));
        li=unli/sum(unli);
        indall=floor(li*Nl)+1;
        for j=1:Ndim
            for k=(j+1):Ndim
                histogram2(indall(j),indall(k))=histogram2(indall(j),indall(k))+1;
            end
        end
    end

    S2_k=log(histogram2);
    G2_k=-S2_k+mean(S2_k(isfinite(S2_k)));

    % plot(dl/2:dl:1,histogram1)
    % plot((dl/2):dl:1,G12_k')

    save(['G2_',num2str(Ndim),'.dat'],'G2_k','-ascii')

end
